api/resolvers/item.js

```diff
@@ -960,6 +961,40 @@ export default {
         sats
       }
     },
+    anonAct: async (parent, { id, sats }, { me, models, lnd }) => {
+      // this should never be called when logged in
+      if (me) {
+        throw new AuthenticationError('you must not be logged in')
+      }
+
+      await ssValidate(amountSchema, { amount: sats })
+
+      const user = await models.user.findUnique({ where: { name: ANON_USER_NAME } })
+
+      // set expires at to 3 hours into future
+      const expiresAt = new Date(new Date().setHours(new Date().getHours() + 3))
+      const description = `Zapping item @${id} on stacker.news`
```

Comment:
we should probably allow anons to hide invoice descriptions

Reply:
Yes their description should be empty by default. We should also probably remove their cowboy hat and remove them from 'top' ... all of which we can do by setting the appropriate fields on the `anon` user record.